jplag.endToEndTesting/src/main/java/de/jplag/end_to_end_testing/helper/JPlagTestSuiteHelper.java

```diff
@@ -1,120 +1,90 @@
- * required plagiarisms are copied from the resource folder to a temporary location, thus creating a folder structure
- * that can be tested by JPlag. Models are instantiated here and required information for the tests is loaded.
- */
 public class JPlagTestSuiteHelper {
 
-    private static final Logger logger = LoggerFactory.getLogger(JPlagTestSuiteHelper.class);
-
-    private String[] resourceNames;
-    private List<JsonModel> resultModel;
-    private LanguageOption languageOption;
-
     /**
-     * Helper class for the endToEnd tests. In this class the necessary resources are loaded, prepared and copied for the
-     * tests based on the passed parameters. An instance of this class loads all necessary paths and properties for a test
-     * run with the specified language
-     * @param languageOption for loading language-specific resources
-     * @throws IOException is thrown for all problems that may occur while parsing the json file. This includes both reading
-     * and parsing problems.
+     * private constructor to prevent instantiation
      */
-    public JPlagTestSuiteHelper(LanguageOption languageOption) throws IOException {
-        this.languageOption = languageOption;
-        this.resourceNames = new File(TestDirectoryConstants.BASE_PATH_TO_JAVA_RESOURCES_SORTALGO.toString()).list();
-
-        this.resultModel = JsonHelper.getResultModelFromPath();
-        logger.debug("temp path at [{}]", TestDirectoryConstants.TEMPORARY_SUBMISSION_DIRECTORY_NAME);
+    private JPlagTestSuiteHelper() {
+        // For Serialization
     }
 
     /**
-     * creates all necessary folder paths and objects for a test run. Also searches for the stored previous results of the
-     * test in order to compare them with the current results.
-     * @param classNames Array of class names with language specific extension to be prepared for a test.
-     * @return comparison results saved for the test
-     * @throws IOException Exception can be thrown in cases that involve reading, copying or locating files.
+     * Loads all existing test data into a test structure. the complex structure consists of the LanguageOption and their
+     * data to be tested. These are divided into folders. for more information please read the README file
+     * @return mapped LanguageOption to the data under test
+     * @throws IOException
      */
-    public TestCaseModel createNewTestCase(String[] classNames, String functionName) throws IOException {
-        createNewTestCaseDirectory(classNames);
-        JsonModel resultJsonModel = resultModel.stream().filter(jsonModel -> functionName.equals(jsonModel.getFunctionName())).findAny().orElse(null);
-        return new TestCaseModel(TestDirectoryConstants.TEMPORARY_SUBMISSION_DIRECTORY_NAME, resultJsonModel, languageOption);
+    public static Map<LanguageOption, Map<String, Path>> getAllLanguageResources() throws IOException {
```

Comment:
Remove throws declarations, as it is no longer required.

Reply:
done